src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -798,7 +798,13 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 		// Find target groups to open
 		const mapGroupToEditorsCandidates = new Map<IEditorGroup, IEditorInputWithOptions[]>();
 		if (group === SIDE_GROUP) {
+			// Custom editors which are open and only support one editor per resource should just move
+			const singletonEditors = this.getEditors(EditorsOrder.SEQUENTIAL).filter(e => typedEditors.find(t => isEqual(e.editor.resource, t.editor.resource)) && !e.editor.canSplit());
```

Comment:
I think the proper way to figure out if an editor matches should be to use it's `matches()` method, or not?

Reply:
This is a tough one.... typedEditors is a set of FileEditorInputs (in the case of open to the side) therefore they will never match the custom editor ones in the `matches` call. I could argue it would work if I put this below the override logic, but the override logic has the logic which closes the additional editor inputs if we only support a single per resource leading to the problem above. 

In both scenarios this bug still exists 
1. Open icon.png in Luna paint
2. Switch to the hex editor using reopen with
3. Open icon.png to the side
4. 🐛 The hex editor moves. I should've gotten Luna paint to the side.

I could investigate how to fix this now, but I believe it's a challenging and complex issue which is too risky this late on. This solution prevents data loss, does something somewhat sensible, and we never supported opening custom editors to the side before.